### ISSUE TITLE
How to work with limitation "Altering a model field from or to AutoField at migration" not supported?

### ISSUE THREAD

**Issue by MSoup** [microsoft/mssql-django#87]
Forgive me if this is a Django issue and not a mssql-django issue. I'm trying to find hints as to why I'm unable to have model instances with ids auto generated by django. 

I have an existing db. I've ran `manage.py inspectdb` and copied the generated models over to models.py. Is the limitation in the title saying that I need to manually comb through these generated models and change intfields to autofields? 

Specifically, I have this model:

```py
class Clerk(models.Model):
    clientid = models.BigAutoField(db_column='clientId', primary_key=True)
    clerkname = models.CharField(db_column='clerkName', max_length=30)
    ...
    class Meta:
        managed = False
        db_table = 'Clerks'
```

clientid is an autofield with primary key set on it, so I feel like it should auto increment with each new `Clerk` instance. 

However, it does not autopopulate or increment:

```py
>>> from webapi.models import Clerk
>>> Clerk.objects.all()
<QuerySet []>
>>> clerk = Clerk(clerkname="hi")
>>> clerk.clientid
```

Because the clientid isn't populated, attempting to save it brings about this error

```py
>>> clerk.save()
Traceback (most recent call last):
  File "C:\git\django-mssql\venv\lib\site-packages\django\db\backends\utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "C:\git\django-mssql\venv\lib\site-packages\mssql\base.py", line 576, in execute
    return self.cursor.execute(sql, params)
pyodbc.IntegrityError: ('23000', "[23000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Cannot insert the value NULL into column 'clientId', table 'web.dbo.Clerks'; column does not allow nulls. INSERT fails. (515) (SQLExecDirectW); [23000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The statement has been terminated. (3621)")

```

And if I set the clientid myself before saving:

```
>>> clerk.clientid=1
>>> clerk.save()
Traceback (most recent call last):
  File "C:\git\django-mssql\venv\lib\site-packages\django\db\backends\utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "C:\git\django-mssql\venv\lib\site-packages\mssql\base.py", line 576, in execute
    return self.cursor.execute(sql, params)
pyodbc.ProgrammingError: ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Table 'Clerks' does not have the identity property. Cannot perform SET operation. (8106) (SQLExecDirectW)")
```

I'm at a bit of a loss. Is this related to the current limitations of mssql-django?

**Comment by absci** [comment]
The limitation is when you alter existing fields from/to Autofield, which is not your case.
But, I couldn't reproduce the issue, clerk.clientid should be None until save called. You shouldn't have to assign a value to clientid. Could you post the original SQL query that create this table? What version of mssql-django you're currently using?

**Comment by MSoup** [comment]
Thank you for your response.

I have since made a lot of progress since my original post. I don't remember what the original SQL query would have been, but the issue was that the MSSQL table, Clerks, did not have clientid set to Identity = True; this meant that clerk.save() would try to save whatever I specified, even if clientid was set to be the primary key. 

The root issue that brought about this error was managed was set to False (which means I was using a preexisting db), but I had made an error when making the DB before connecting Django to it. 

mssql-django works as expected!
